### PR TITLE
API specification changes for custom roles 

### PR DESCRIPTION
**Pull request by n1v0lg**
With changes for Serverless custom roles, several roles-related APIs have stricter request validation. This PR codifies these requirements into our ES API specification. 

Notes & questions to reviewers:

* The changes to the Elasticsearch APIs are **still behind a feature flag** (not a proper one, but rather a setting). I should only merge this spec PR once the feature flag is removed, i.e., once the APIs are truly public, correct?
* The change introduced quite a few validation failures. I believe they are related to the changes in this PR, but I'm unable to run local validation to debug these, due to local setup issues (note my message in the clients channel -- not linking to avoid a private link in a public repo).

Closes: ES-7833

### CODE REVIEW

**Comment by n1v0lg** [comment]
@elasticmachine update branch

File: specification/security/_types/Privileges.ts [hunk 1]
```diff
@@ -42,40 +42,100 @@ export class ApplicationPrivileges {
...
   monitor_ml,
+  /**
+   * @availability stack
+   */
   monitor_rollup,
+  /**
+   * @availability stack
+   */
   monitor_snapshot,
   monitor_text_structure,
```
Comment (pquentin) [inline specification/security/_types/Privileges.ts:117]: I'm not seeing this one in https://github.com/elastic/elasticsearch-serverless/blob/c093830efcb3e2873e2307bb7b3da67d3966e768/modules/serverless-security/src/main/java/co/elastic/elasticsearch/serverless/security/privilege/ServerlessSupportedPrivilegesRegistry.java

Comment (n1v0lg) [reply specification/security/_types/Privileges.ts:117]: Great catch, I'll add a `@availability stack` here too. Thanks!

**Comment by n1v0lg** [comment]
> Would you mind opening a follow-up pull request to add the missing ClusterPrivilege (like NONE or MANAGE_BEHAVIORAL_ANALYTICS) and IndexPrivilege (like MANAGE_DATA_STREAM_LIFECYCLE or CROSS_CLUSTER_REPLICATION) values? I suppose it can go in this one too, if you prefer.

For sure. I'll make it a separate PR because we can make these changes now already, whereas the current PR will have to wait until the Serverless-side API changes are ready to be made public.

**Review by pquentin — APPROVED** [review]

**Pull request merged**